### PR TITLE
Use annotation class name matching instead of the actual class to avoid license issues.

### LINKED ISSUE #1360: Remove jsr-305 dependency
> Is their no better way to achieve the same without JSR 305?

check the annotation classname rather than the class itself.  (which is what spotbugs acutally does to support a viarety of `Nonnull` classes.

re-introducing jsr-305 goes against the point of getting rid of it as it has a horrible license.

now the annotation won't be present unless the annotation class can be loaded (and this may well cause you not to find the annotations even if you provide it - if the classloader of the class with the use of the annotation can not load the class via itself or its parents.  So what you are trying to achieve here is probably doomed to failure anyway (or the other plugin will have the class but it will be a different one due to the classloader and then probably not match when you expect it would - and your unit tests give false positives due to the Classloading differences in Jenkins vs JenkinsRule)

_Originally posted by @jtnord in https://github.com/jenkinsci/configuration-as-code-plugin/pull/1359#issuecomment-614102100_

### PR DESCRIPTION
**Pull request by jetersen**
fixes #1360 

Not sure if there is a better way to encapsulate the method to extract the annotations for the different classes (`Package`, `Parameter`, `AccessibleObject`, `Class<T>`)

### Your checklist for this pull request

🚨 Please review the [guidelines for contributing](../blob/master/docs/CONTRIBUTING.md) to this repository.

- [ ] Make sure you are requesting to **pull a topic/feature/bugfix branch** (right side) and not your master branch!
- [ ] Ensure that the pull request title represents the desired changelog entry
- [ ] Please describe what you did
- [ ] Link to relevant issues in GitHub or in [Jenkins JIRA](https://issues.jenkins-ci.org)
- [ ] Link to relevant pull requests, esp. upstream and downstream changes
- [ ] Did you provide a test-case? That demonstrates feature works or fixes the issue.

### CODE REVIEW

File: test-harness/pom.xml [hunk 1]
```diff
@@ -73,6 +73,13 @@
       <artifactId>system-rules</artifactId>
       <version>1.19.0</version>
     </dependency>
+
+    <dependency>
```
Comment (timja) [inline test-harness/pom.xml:77]: why is this needed here?

Comment (jetersen) [reply test-harness/pom.xml:77]: it is I simply forgot to remove it after moving it thinking I would need it.

File: test-harness/pom.xml [hunk 2]
```diff
@@ -73,6 +73,13 @@
...
+      <groupId>com.google.code.findbugs</groupId>
+      <artifactId>jsr305</artifactId>
+      <version>3.0.2</version>
+      <scope>test</scope>
+    </dependency>
```
Comment (jetersen) [inline test-harness/pom.xml:82]:
```suggestion
```

**Comment by jetersen** [comment]
actually this is not possible with the following to two cases as SpotBugs annotations are not compiled into runtime.
So we have two failing test cases `CheckForNull` (PackageParametersNonNullCheckForNull) and `NonNull` (NonnullParameterConstructor)

I am not sure how to solve this without usage of Jsr305...

**Comment by timja** [comment]
> actually this is not possible with the following two test cases as SpotBugs annotations are not compiled into runtime.
> So we have two failing test cases `CheckForNull` (PackageParametersNonNullCheckForNull) and `NonNull` (NonnullParameterConstructor)
> 
> I am not sure how to solve this without usage of Jsr305...

Isn't the intention to use jsr305 but by name only without a compile dependency?

**Comment by jetersen** [comment]
PackageParametersNonNullCheckForNull and NonnullParameterConstructor tests depend on jsr305 to be compiled to work.

**Comment by timja** [comment]
O for the tests? Could we just include in tests?

**Comment by jetersen** [comment]
I won't work in production code either if we have enforced everyone to use spotbugs annotations.
